cloud/scope/vnic_reconciler.go

```diff
@@ -0,0 +1,142 @@
+package scope
+
+import (
+	"context"
+	"fmt"
+
+	infrastructurev1beta1 "github.com/oracle/cluster-api-provider-oci/api/v1beta1"
+	"github.com/oracle/cluster-api-provider-oci/cloud/ociutil"
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-go-sdk/v65/core"
+	"github.com/pkg/errors"
+)
+
+func (m *MachineScope) ReconcileVnicAttachments(ctx context.Context) error {
+	if m.IsControlPlane() {
+		return errors.New("cannot attach multiple vnics to ControlPlane machines")
+	}
+
+	for index, vnicAttachment := range m.OCIMachine.Spec.VnicAttachments {
+		if m.vnicAttachmentExists(ctx, vnicAttachment) {
```

Comment:
this will be an infinite loop if we dont do comparison right?

Reply:
updated